typescript/style.yml

```diff
@@ -0,0 +1,136 @@
+# Copyright 2019 OpenStax Poland
+# Licensed under the MIT license. See LICENSE file in the project root for
+# full license text.
+
+extends: ./base.yml
+
+rules:
+  # Prefer type[] over Array<type>.
+  '@typescript-eslint/array-type': error
+
+  # Require all class members to be explicitly marked as either public,
+  # protected, or private.
+  '@typescript-eslint/explicit-member-accessibility': error
+
+  # Require non-inferred type annotations on all public APIs. For now warnings
+  # only, as this is a new lint and we aren't sure how well it works. In future
+  # we'd like to turn in into hard error.
+  '@typescript-eslint/explicit-module-boundary-types': warn
+
+  # Don't use member delimiters in interface declarations (for consistency with
+  # class declarations) and comma in type declarations (for consistency with
+  # object literals).
+  '@typescript-eslint/member-delimiter-style':
+    - error
+    - multiline:
+        delimiter: none
+      singleline:
+        delimiter: comma
+      overrides:
+        typeLiteral:
+          multiline:
+            delimiter: comma
+            requireLast: true
+
+  # Enforce consistent naming convention. Since this is a new lint, and defining
+  # a style guide for TS/JS is non-trivial, this is currently only a warning.
+  '@typescript-eslint/naming-convention': warn
+
+  # Disallow unnecessary non-null assertions.
+  '@typescript-eslint/no-extra-non-null-assertion': error
+
+  # Disallow usage of classes as namespaces.
+  '@typescript-eslint/no-extraneous-class': error
+
+  # Forbid comparing against a boolean literal.
+  '@typescript-eslint/no-unnecessary-boolean-literal-compare': error
+
+  # Forbid unnecessary type assertions.
+  '@typescript-eslint/no-unnecessary-type-assertion': error
+
+  # Prefer `literal as const` over `literal as literal`.
+  '@typescript-eslint/prefer-as-const': error
+
+  # Prefer for-of loops over looping with an index.
+  '@typescript-eslint/prefer-for-of': error
```

Comment:
We're already transpiling for [ES 2017](https://github.com/openstax-poland/adaptarr-front/blob/master/tsconfig.json#L6) so this is not a problem for us. And besides, I think that a few extra bytes in final script is a good price to pay for more readable and maintainable code.

Reply:
In Adaptarr, yes, but this config is meant to be used by more than one repo

Reply:
Do we have any other projects where use a lower target? Do we expect to ever want to support a lower target? And even if, my second point still stands.

Reply:
> I think that a few extra bytes in final script is a good price to pay for more readable and maintainable code.

True, I've misunderstood the docs

About this lint rule:
I don't think it should be set to an error, because using a standard for loop is a valid solution and it will not cause errors on production (referring to the https://github.com/openstax-poland/eslint-config-openstax-poland/pull/4#discussion_r447665790)